docs/src/content/docs/request-state.mdx

```diff
@@ -0,0 +1,59 @@
+---
+title: Request State
+description: Request State
+packageName: '@inox-tools/request-state'
+challenger: [Florian, Oliver]
+---
+
+Astro provides [`Astro.locals`/`context.locals`](https://docs.astro.build/en/reference/api-reference/#astrolocals) for shared state between different components being rendered in a page.
+
+This state can only be accessed from an Astro component using the `Astro` constant or from middlewares from the context. Sharing state between framework components is not provided and inconvenient.
+
+Framework components using the shared state won't have the same state on the client, resulting in problems with the server and the client rendering different contents.
+
+This library provides a solution for that, any state set using this library is shared across any component (even from UI frameworks) used in rendering the page in both server and client side.
+
+## Installing the dependency
+
+import { PackageManagers } from 'starlight-package-managers';
+
+<PackageManagers type="exec" pkg="astro" args="add @inox-tools/request-state" />
+
+## How to use
+
+Import the two functions anywhere in your code:
+
+```ts
+import { getState, setState } from '@it-astro:state';
+
+setState('some key', 'some value');
+
+const state = getState('some key');
+```
+
+The state can be any value supported by the [`devalue` library](https://www.npmjs.com/package/devalue) plus:
+
+- `Date`s
+- `URL`s
+- Global symbols (`Symbol.for`)
+- Well-known symbols (`Symbol.XXX`)
+
+### `setState`
+
+Params:
+
+- `key` (`string`): The key of a value in the state
+- `value` (`unknown`): The value to be set
+
+### `getState`
+
+Params:
+
+- `key` (`string`): The key of a value in the state
+- `valueIfMissing` (`unknown`): An optional value to set if there is no value in the state for the given key.
```

Comment:
I don't know if it's a good idea. I think it would be better to recommend creating custom hooks like for react context so you can error/provide a default as a user

Reply:
I don't get the idea here

Reply:
basically i don't see the value of `getState('x', 'DEFAULT')` over `getState('x') ?? 'DEFAULT'`

Reply:
Oh, the difference is that the value will be set on the state if it's not there. So the equivalent would be this:
```ts
let value = getState('key');
if (value === undefined) {
  value = initialValue;
  setState('key', value);
}
```

Reply:
Ah I see 👍